batch_job/src/work_queue_factory.c

```diff
@@ -373,6 +375,14 @@ static int submit_worker( struct batch_queue *queue )
 		free(files);
 		files = newfiles;
 	}
+	
+	if(os){
+		char* temp = string_format("python /afs/crc.nd.edu/group/ccl/software/runos/runos.py %s %s",os,cmd);
```

Comment:
I think you want to have runos in your PATH, copy it to scratch_dir, and send it as we send the worker. (See the line that has "which work_queue_worker".